From 1aaa37b35b303f092931bfeedbec20cedd251eae Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 4 Mar 2016 17:26:47 -0500
Subject: [PATCH 0878/1936] MiddleWare: Handle Missing PAD Synths

This corrects a possible data race when loading an instrument while
a pad window is open.
---
 src/Misc/MiddleWare.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 4761daec..a431def8 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -349,7 +349,8 @@ struct NonRtObjStore
             d.matches++;
             d.reply((obj_rl+"needPrepare").c_str(), "F");
         } else {
-            assert(pad);
+            if(!pad)
+                return;
             strcpy(d.loc, obj_rl.c_str());
             d.obj = pad;
             PADnoteParameters::non_realtime_ports.dispatch(msg, d);
-- 
2.47.0

